pkg/scheduler/core/generic_scheduler.go

```diff
@@ -237,6 +241,17 @@ func (g *genericScheduler) selectHost(nodeScoreList framework.NodeScoreList) (st
 	return selected, nil
 }
 
+// ResolveFilterErr tries to resolve <err> by running registered PostFilter plugins.
+func (g *genericScheduler) ResolveFilterErr(ctx context.Context, prof *profile.Profile, state *framework.CycleState, pod *v1.Pod, err error) {
+	filterErr, ok := err.(*FitError)
+	if !ok || filterErr == nil {
+		return
+	}
```

Comment:
PostFilter plugins should run irrespective of the error type I think, so I expect this check to happen inside the preemption PostFilter plugin.

Reply:
> PostFilter plugins should run irrespective of the error type I think

Do you expect PostFilter to run upon just failure of Filter, or running upon both failure and success of Filter phase?

Reply:
I think it should always run, just like the old semantics

Reply:
The plugin itself can decide whether or not to actually do something

Reply:
The old semantics treats PostFilter as an information extension point after the **success** of Filter phase - in other words, the old PostFilter doesn't run in the failure path after Filter. So now I suppose we're going to reserve that semantics, and also extend it to run upon the **failure** of Filter?

Reply:
Hmm, IIRC the old semantics of postfilter always run, on both success and failure

Reply:
Ah true, it used to run right after Filter:

https://github.com/kubernetes/kubernetes/blob/d32e40e20d167e103faf894261614c5b45c44198/pkg/scheduler/core/generic_scheduler.go#L218-L219

Reply:
@ahg-g commit "fixup: run PostFilter plugins right after Filter phase" did the change.